assets/blocks/single-lesson.js

```diff
@@ -17,3 +23,32 @@ registerSenseiBlocks( [
 	ResetLessonBlock,
 	ViewQuizBlock,
 ] );
+
+let needsTemplate;
```

Comment:
I think this might not be the right direction for this. Could we just add the post type template as the `post_content` of `wp_insert_post` in the [create_lesson](https://github.com/Automattic/sensei/blob/e908725fe2a8d5695e08063c7f7708694610ad56/includes/class-sensei-course-structure.php#L456) method?

Reply:
Nice! I'll update to this method tomorrow.
And I thought adding it when the content is empty, but it can cause weird behaviors if the user just decide to clear the post type and refresh the page.

Reply:
I think we can live with that, I don't think it's common that they just clear the post and leave. It's a small price to pay for avoiding the extra complexity of having a meta for this.

Reply:
Hum, I don't think it's a considerable complexity to keep the "original behavior". But, of course, if we don't need that, it would be good to remove it. WDYT @donnapep?

And I updated the action here: https://github.com/Automattic/sensei/pull/3937/commits/0e65233607021a389574585367a00bb303ecc8e8. Thank you for the suggestion, @yscik! =)

Reply:
It's not ideal, but if it adds complexity or takes significant time to fix, then probably not worth it since this isn't likely isn't an issue that will be encountered too often.

What troubles me a bit more is that the _View Quiz_ button is selected in the editor by default, so you see the toolbar in the editor. Whereas if you create a new post or page, the focus is on the title field. Is that easy to fix?

Reply:
I mean the extra complexity in the code of creating, querying and updating a meta field just for tracking this. It also wouldn't solve the problem if the lesson post was created by something else, like an extension.

Reply:
I'm trying to remove the meta and implement the idea of the post empty, but it's being more complicated because we don't know when the editor already added the blocks.

I'm still debugging the store and trying to find if there is something changed in the state when it's ready, but for now it seems there is no clue for that.

Reply:
I explored a lot how to implement that solution, but it's hard. 🥵

Basically, what happens is that the initial state is like we don't have any block in the editor. The values are not `null`, or `undefined` until to be initialized. They are `false`, `[]`, `0`, and so on. And we don't have any selector that gives us a clue that the `resetBlocks` is already done.

So, for now, I suggest we continue with this as it is. And if in some moment we find a better solution, we can implement that.

Reply:
> It's the common behavior when editing a post, so I think we could keep it as it's, right?

Sure, that makes sense. Tested this PR and all looks good.